jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/CompilationMXBeanImpl.java

```diff
@@ -114,6 +113,9 @@ public boolean isCompilationTimeMonitoringSupported() {
 	 */
 	@Override
 	public ObjectName getObjectName() {
+		if (objectName == null) {
+			objectName = ManagementUtils.createObjectName(ManagementFactory.COMPILATION_MXBEAN_NAME);
+		}
```

Comment:
is there a common superclass where we can put this implementation so it's not duplicated in every bean?  Doing something like that would require all bean's to add their poolName when created (instance field) or to override an abstract "getPoolName" method

Reply:
Most beans extend `Object`, so there's no common superclass.
How is overriding something like `getObjectNameString()` better than overriding `getObjectName()`?

Reply:
I had been hoping for a common super class so we could do: 
```
class BeanSuper {

public ObjectName getObjectName() {
  if (objectName) == null) {
     objectName = ManagementUtils.createObjectName(getObjectNameString());
  }
}

abstract String getObjectNameString();
}
```
`getObjectNameString()` is a much clearer contract than knowing to delay the initialization of the ObjectName like this.

Reply:
I'm not convinced that's a better outcome.

In any case, there are obstacles:

1. `MemoryMXBeanImpl extends LazyDelegatingNotifier implements MemoryMXBean`; the notifier class shouldn't provide `getObjectName()`

2. `MemoryPoolMXBeanImpl` creates its object name from two strings (a domain *and* a name)

Reply:
Ok, let's drop this suggestion then as it's not practical given the existing MxBean code.  

I strongly dislike this kind of copy/paste as it's likely to be missed the next time someone implements a bean